From 9f0ae412b7b82212b6ea912b8adedbaa4128d120 Mon Sep 17 00:00:00 2001
From: Walter Su <waltersu4549@apache.org>
Date: Mon, 23 Nov 2015 15:35:06 +0800
Subject: [PATCH 1522/2748] HDFS-9428. Fix intermittent failure of
 TestDNFencing.testQueueingWithAppend. Contributed
 by Masatake Iwasaki.

(cherry picked from commit 5aba093361dcf6bb642e533700f772b9a94154ad)
(cherry picked from commit 846de7d8429c9e071287384802c0af81f868e1ed)

Change-Id: Ia6f7f4914cb03c2a7db74bb821a358beaa4f4a7d
---
 .../hdfs/server/namenode/ha/TestDNFencing.java     |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestDNFencing.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestDNFencing.java
index e915828..3daad7a 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestDNFencing.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestDNFencing.java
@@ -457,6 +457,7 @@ public void testQueueingWithAppend() throws Exception {
       numQueued += numDN * 2; // RBW messages, see comments in case 1
     } finally {
       IOUtils.closeStream(out);
+      cluster.triggerHeartbeats();
       numQueued += numDN; // blockReceived
     }
     assertEquals(numQueued, cluster.getNameNode(1).getNamesystem().
-- 
1.7.9.5

